src/login.ts

```diff
@@ -0,0 +1,242 @@
+/**
+ * The type of the login challenge.
+ */
+const CHALLENGE_TYPE_LOGIN = "skynet-portal-login";
+/**
+ * The type of the registration challenge.
+ */
+const CHALLENGE_TYPE_REGISTER = "skynet-portal-register";
+
+/**
+ * Custom register options.
+ *
+ * @property [endpointRegisterRequest] - The relative URL path of the portal endpoint to contact.
+ * @property [endpointRegister] - The relative URL path of the portal endpoint to contact for large uploads.
+ */
+export type CustomRegisterOptions = CustomClientOptions & {
+  endpointRegister?: string;
+  endpointRegisterRequest?: string;
+};
+
+/**
+ * Custom login options.
+ *
+ * @property [endpointLoginRequest] - The relative URL path of the portal endpoint to contact.
+ * @property [endpointLogin] - The relative URL path of the portal endpoint to contact for large uploads.
+ */
+export type CustomLoginOptions = CustomClientOptions & {
+  endpointLogin?: string;
+  endpointLoginRequest?: string;
+};
+
+/**
+ * The default custom client options.
+ */
+const DEFAULT_CUSTOM_CLIENT_OPTIONS = {
+  APIKey: "",
+  customUserAgent: "",
+  customCookie: "",
+  onDownloadProgress: undefined,
+  onUploadProgress: undefined,
+};
+
+export const DEFAULT_REGISTER_OPTIONS = {
+  ...DEFAULT_CUSTOM_CLIENT_OPTIONS,
+
+  endpointRegisterRequest: "/api/register",
+  endpointRegister: "/api/register",
+};
+
+export const DEFAULT_LOGIN_OPTIONS = {
+  ...DEFAULT_CUSTOM_CLIENT_OPTIONS,
+
+  endpointLoginRequest: "/api/login",
+  endpointLogin: "/api/login",
+};
+
+/**
+ * The challenge response.
+ *
+ * @property response - The hex-encoded byte array of the signed data, e.g. challenge+type+recipient. The type is either
+ * `skynet-portal-login` or `skynet-portal-register`, depending on the endpoint on which the challenge was requested.
+ * @property signature - The signature of the data.
+ */
+type ChallengeResponse = {
+  response: string;
+  signature: string;
+};
+
+/**
+ * Generates a portal login keypair.
+ *
+ * @param seed - The user seed.
+ * @param email - The email.
+ * @returns - The login keypair.
+ */
+function genPortalLoginKeypair(seed: Uint8Array, email: string): KeyPair {
+  const hash = hashWithSalt(seed, email);
+
+  return genKeyPairFromHash(hash);
+}
+
+/**
+ * Registers a user for the given seed and email.
+ *
+ * @param client - The Skynet client.
+ * @param seed - The seed.
+ * @param email - The user email.
+ * @param [customOptions] - The custom register options.
+ * @returns - The JWT token.
+ */
+export async function register(
+  client: SkynetClient,
+  seed: Uint8Array,
+  email: string,
+  customOptions?: CustomRegisterOptions
+): Promise<string> {
+  const opts = { ...DEFAULT_REGISTER_OPTIONS, ...client.customOptions, ...customOptions };
+
+  const { publicKey, privateKey } = genPortalLoginKeypair(seed, email);
+
+  const registerRequestResponse = await client.executeRequest({
+    endpointPath: opts.endpointRegisterRequest,
+    method: "GET",
+    subdomain: "account",
```

Comment:
`subdomain` is not part of `RequestConfig`.

Reply:
Updated skynet-js version.